### PR TITLE
# Remove unused fields and resolve NPE

### PR DESCRIPTION
**Pull request by immmus**
I've removed a lot of unused code since 2018. I also moved the repetitive logic to the main class. Resolved NPE issue after last update.

### CODE REVIEW

File: open-metadata-implementation/access-services/subject-area/subject-area-api/pom.xml [hunk 1]
```diff
@@ -6,7 +6,6 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <groupId>org.odpi.egeria</groupId>
```
Comment (davidradl) [inline open-metadata-implementation/access-services/subject-area/subject-area-api/pom.xml:9]: I think it best o leave this line in - so it is explicit unless @planetf1 thinks it is ok.

**Review by davidradl — COMMENTED** [review]
Looks good to me -  I just wanted to check the pom change with Nigel @planetf1

File: open-metadata-implementation/access-services/subject-area/subject-area-api/pom.xml (hunk 1 above)
Comment (immmus) [reply open-metadata-implementation/access-services/subject-area/subject-area-api/pom.xml:9]: This is a line and so is explicitly specified in the parent tag;)
Screenshot 2020-09-01 at 11 15 56

Comment (planetf1) [reply open-metadata-implementation/access-services/subject-area/subject-area-api/pom.xml:9]: I agree with @immmus - there is no need to have this groupId in (though should be harmless). Our other poms typically do not include it.

**Review by planetf1 — APPROVED** [review]

**Review by davidradl — APPROVED** [review]

**Pull request merged**